### PR TITLE
PWN-7382 - Add Receive Ethereum Fragment

### PR DESCRIPTION
**Pull request by Davran331122**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7382

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/receive/eth/ReceiveEthereumFragment.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import org.p2p.wallet.utils.copyToClipBoard
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.viewbinding.viewBinding
+import org.p2p.wallet.utils.withArgs
+
+private const val ARG_TOKEN_LOGO_URL = "ARG_TOKEN_LOGO_URL"
+private const val ARG_TOKEN_SYMBOL = "ARG_TOKEN_SYMBOL"
+
+class EthereumReceiveFragment :
+    BaseMvpFragment<ReceiveEthereumContract.View,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/receive/eth/ReceiveEthereumFragment.kt:22]: strange formatting, maybe you can you typealias to make it more readable
or static imports

File: app/src/main/java/org/p2p/wallet/receive/eth/ReceiveEthereumFragment.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        ReceiveEthereumContract.Presenter>(
+        R.layout.fragment_ethereum_receive
+    ),
+    ReceiveEthereumContract.View {
+
+    companion object {
+        fun create(tokenSymbol: String, tokenLogo: String) = EthereumReceiveFragment().withArgs(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/receive/eth/ReceiveEthereumFragment.kt:29]: explicit return type

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -337,11 +337,19 @@
     <string name="receive_pay_and_countinue">Pay %s SOL &amp; Continue</string>
     <string name="receive_on_solana">%s on Solana</string>
     <string name="receive_my_sol_address">My Solana address</string>
+    <string name="receive_my_eth_address">My Ethereum address</string>
     <string name="receive_my_username">My username</string>
     <string name="receive_copy_address">Copy address</string>
     <string name="receive_sol_address_copied">✅ Your Solana address was copied</string>
     <string name="receive_eth_address_copied">✅ Your Ethereum address was copied</string>
     <string name="receive_username_copied">✅ Your username was copied</string>
+    <string name="receive_on_ethereum">%s on Ethereum</string>
```
Comment (gslevinkov) [inline app/src/main/res/values/strings.xml:346]: you can try and extract receive strings to separate strings.xml like we did in swap

File: ethereumkit/src/test/java/org/p2p/ethereumkit/TokenBalancesTest.kt [hunk 4]
```diff
@@ -40,6 +40,7 @@ class TokenBalancesTest : KoinTest {
     fun getBalance() = runTest {
         repository.init(seedPhrase = "apart approve black comfort steel spin real renew tone primary key cherry".split(" "))
         val result = repository.loadWalletTokens()
+        println(result)
```
Comment (gslevinkov) [inline ethereumkit/src/test/java/org/p2p/ethereumkit/TokenBalancesTest.kt:43]: bruh

**Review by gslevinkov — APPROVED** [review]

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/receive/eth/ReceiveEthereumFragment.kt [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        super.onViewCreated(view, savedInstanceState)
+        with(binding) {
+            toolbar.title = getString(R.string.receive_on_ethereum, tokenSymbol)
+            toolbar.setNavigationOnClickListener { popBackStack() }
+            glideManager.load(imageViewWatermark, logoUrl)
+        }
+        presenter.load()
+    }
+
+    override fun showQrAndAddress(qrBitmap: Bitmap, addressInHexString: String) {
```
[the remark this answers is not in the document]
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/receive/eth/ReceiveEthereumFragment.kt:52]: thanks!

File: ethereumkit/src/test/java/org/p2p/ethereumkit/TokenBalancesTest.kt (hunk 4 above)
Comment (Davran331122) [reply ethereumkit/src/test/java/org/p2p/ethereumkit/TokenBalancesTest.kt:43]: Avada kevabra ----->

File: app/src/main/res/layout/fragment_ethereum_receive.xml [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+                        android:layout_gravity="center"
+                        android:src="@drawable/ic_sol"
+                        app:cornerRadius="16dp" />
+
+                </FrameLayout>
+
+            </FrameLayout>
+
+            <LinearLayout
+                android:id="@+id/containerAddress"
```
Comment (NikolaiEmelianov) [inline app/src/main/res/layout/fragment_ethereum_receive.xml:67]: FinanceBlock in figma

**Review by NikolaiEmelianov — CHANGES_REQUESTED** [review]

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/recycler/decoration/GroupedRoundingDecoration.kt [hunk 7]
```diff
@@ -19,7 +19,7 @@ import org.p2p.uikit.utils.drawable.shape.shapeTopRounded
 import org.p2p.uikit.utils.recycler.getItems
 import org.p2p.uikit.utils.toPx
 
-fun groupedRoundingFinanceBlockDecoration(@Px round: Float = 12f.toPx()) =
+fun groupedRoundingFinanceBlockDecoration(@Px round: Float = 16f.toPx()) =
     GroupedRoundingDecoration(FinanceBlockCellModel::class, round)
 
 class GroupedRoundingDecoration(
```
Comment (NikolaiEmelianov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/utils/recycler/decoration/GroupedRoundingDecoration.kt:25]: and constructor 16f

File: ui-kit/src/main/kotlin/org/p2p/uikit/utils/recycler/decoration/RoundingByCellDecoration.kt [hunk 8]
```diff
@@ -15,7 +15,7 @@ import org.p2p.uikit.utils.drawable.shape.shapeRoundedAll
 import org.p2p.uikit.utils.recycler.getItems
 import org.p2p.uikit.utils.toPx
 
-fun roundingByCellFinanceBlockDecoration(@Px round: Float = 12f.toPx()) =
+fun roundingByCellFinanceBlockDecoration(@Px round: Float = 16f.toPx()) =
     RoundingByCellDecoration(FinanceBlockCellModel::class, round)
 
 class RoundingByCellDecoration(
```
Comment (NikolaiEmelianov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/utils/recycler/decoration/RoundingByCellDecoration.kt:21]: and constructor 16f

**Review by NikolaiEmelianov — APPROVED** [review]

**Pull request merged**